server/db/event-db-query.js

```diff
@@ -0,0 +1,51 @@
+const { Client } = require("pg");
+const { dbQuery } = require("./db-query");
+
+const logQuery = (statement, parameters) => {
+  const timeStamp = new Date();
+  const formattedTimeStamp = timeStamp.toString().substring(4, 24);
+  console.log(formattedTimeStamp, statement, parameters);
+};
+
+module.exports = {
+  async eventDbQuery(statement, ...parameters) {
+    const selectQuery = `SELECT * FROM connection;`;
+    const queryResult = await dbQuery(selectQuery);
+    const credentials = queryResult.rows[0];
+
+    const client = new Client({
+      user: credentials.pg_user,
+      host: credentials.pg_host,
+      password: credentials.pg_password,
+      database: credentials.pg_database,
+      port: credentials.pg_port,
+    });
+
+    await client.connect();
+
+    logQuery(statement, parameters);
+    const result = await client.query(statement, parameters);
+
+    await client.end();
+
+    return result;
+  },
+
+  async verifyConnection({ user, host, password, database, port }) {
+    const client = new Client({
+      user,
+      host,
+      password,
+      database,
+      port,
+    });
+
+    await client.connect();
+
+    console.log(
+      `Connection to host: ${host} database: ${database} successfully verified.`
```

Comment:
How do we know if the connection is unsuccessful? What's the behavior of an unsuccessful `connect()`?

Reply:
Yeah, this took me a while to figure out and I'm not 100% sure my solution is the best option.

Firstly, it's assumed that all the error handling for the methods in `db-query` and `event-Db-Query` are handled in the route where they're called. If you look at the routes Sean created, this is the case. 

So the error handling in this case is contained in the `connectionController` file when the query is called. But, like you said, how do we differentiate the errors? In the catch block on lines 22-27 of `connectionController` I ended up accessing a property in the error object to check what kind of error it is. That's what determines what message is sent back to the client.

Again, not 100% sure if this is the best way to do it but it worked well when I tested it.

Reply:
I see, I think I'm still confused about this exact part since there's no query being run here, so based on my understanding those error handlers won't be reached. I was wondering if we should maybe add a SELECT query between `await client.connect()` and the success message?

Reply:
When `client.connect()` on line 43 of `event-db-query` is run, either an error will be thrown, or it will be successful. I thought about trying to do a query here, but an error would be thrown before I could even query anything.